test/integration/logic/resolvable/UnifyRelationConcludableTest.java

```diff
@@ -787,6 +856,314 @@ public void relation_more_players_than_rule_relation_fails_unify() {
         assertEquals(expected, result);
     }
 
+    @Test
+    public void binaryRelationWithRoleHierarchy_ParentWithBaseRoles(){
+        String parentRelation = "{ (employer: $x, employee: $y); }";
+        String conclusion = "(part-time-employer: $u, part-time-employee: $v) isa part-time-employment";
+        String conclusion2 = "(taxi: $u, night-shift-driver: $v) isa part-time-driving";
+
+        verifyUnificationOutcome(parentRelation, conclusion, "{ $u isa part-time-organisation; $v isa student;}");
```

Comment:
i think it'll be easier to follow if you introduce a `rule(conclusion, condition)` method, which takes the two strings. Then all your "tester" methods will take a `parentRelation`/conjunction and a `Rule` instance. You can get the conjunction out of a rule too